anchor/network/src/handshake/behaviour.rs

```diff
@@ -0,0 +1,313 @@
+use discv5::libp2p_identity::Keypair;
+use discv5::multiaddr::Multiaddr;
+use libp2p::core::transport::PortUse;
+use libp2p::core::Endpoint;
+use libp2p::request_response::{
+    self, Behaviour, Config, Event, OutboundRequestId, ProtocolSupport,
+};
+use libp2p::swarm::{
+    ConnectionDenied, ConnectionId, FromSwarm, NetworkBehaviour, THandler, THandlerInEvent,
+    THandlerOutEvent, ToSwarm,
+};
+use libp2p::{PeerId, StreamProtocol};
+use prost::Message;
+use std::collections::HashMap;
+use std::error::Error;
+use std::sync::{Arc, Mutex};
+use std::task::{Context, Poll};
+use std::time::Instant;
+use tracing::debug;
+use crate::handshake::codec::EnvelopeCodec;
+use crate::handshake::record::envelope::Envelope;
+use crate::handshake::record::signing::{consume_envelope, seal_record};
+use crate::handshake::types::NodeInfo;
+
+/// Event emitted on handshake completion or failure.
+#[derive(Debug)]
+pub enum HandshakeEvent {
```

Comment:
nit:
```suggestion
pub enum Event {
```
the module already gives out the namespace, no need for redundancy

Reply:
done